compiler/noirc_evaluator/src/ssa/opt/constant_folding.rs

```diff
@@ -1960,4 +1947,87 @@ mod test {
         }
         ");
     }
+
+    #[test]
+    fn unknown_issue_regression_test() {
+        // Regression test for an issue discovered in https://github.com/AztecProtocol/aztec-packages/pull/14492
+        // Marking `f1` as `predicate_pure` instead of `impure` results in an invalid storage write.
```

Comment:
Can this name be more descriptive?
Do we still not know what went wrong or is it definitely the fact we were deduplicating a function call without an inc_rc?

Reply:
Ah, sorry just realized you didn't ask for a review yet